src/compiler/utilities.ts

```diff
@@ -134,10 +158,26 @@ module ts {
             return true;
         }
 
-        return node.pos === node.end && node.kind !== SyntaxKind.EndOfFileToken;
+        return node.pos === node.end && node.kind !== SyntaxKind.EndOfFileToken && node.pos >= 0;
+    }
+
+    export function nodeIsGenerated(node: Node) {
+        if (!node) {
+            return true;
+    }

```

Comment:
odd curly.  Also, i don't think an undefined node should be considered 'generated'...